### ISSUE TITLE
How to pass multi-value (IN clause) to filters dynamically?

### ISSUE THREAD

**Issue by JasperSui** [cube-js/cube.js#3291]
I've asked the same question on Slack, but I think there might be someone who will face the same question because it's a typical scenario on reporting.

**Problem**

How to pass multi-value (IN clause) to filters dynamically?

```SQL
SELECT * FROM table
WHERE table.id IN (1, 2, 3, 4, 5)
```

`(1, 2, 3, 4, 5)` will be passed from API or somewhere, so it's mutable.

**Description**

As @rchkv said on Slack, we can write the multi-value in `Cube.sql` or `Cube.filters` directly.

Like
```
cube(`Users`, {
  sql: `SELECT * FROM public.users where ${CUBE}.age in (19)`
})
```

Or
```


cube(`Users`, {
  sql: `SELECT * FROM public.users where {$CUBE}.age in (19)`,
  
  dimensions: {
    age: {
      sql: `age`,
      type: `number`
    }
  },

  segments: {
    ages: {
      sql: `${CUBE}.age in (25, 23)`,
    },
  }
});
```

But it seemed like we need to pre-define the filter or segments at build time?
What I need is to filter by dynamic multi-value which would be passed from API (I don't know is it reachable currently lol..)

Simple demo:

API: `http://service/report?age=23,24,25`
SQL: `SELECT * FROM public.users where users.age in (23,24,25)`

**Comment by JasperSui** [comment]
I just read the [API Doc](https://cube.dev/docs/rest-api#api-reference-v-1-load),
and I realize that what I exactly need is an `IN` operator for filters (which supports both string type and number type).

Demo:
```bash
curl \
 -X POST
 -H "Content-Type: application/json" \
 -H "Authorization: EXAMPLE-API-TOKEN" \
 -G \
 --data '{"query": {"measures":["Users.count"], "filters":[{"users":"users.age","operator":"in","values":[23,24,25]}]}}' \
 http://localhost:4000/cubejs-api/v1/load
```

`"filters":[{"users":"users.age","operator":"in","values":[23,24,25]}]` will be useful if we can pass the value from somewhere than pre-defining the value in .js file.

**Comment by JasperSui** [comment]
I tried this on cube.js develop playground and the frontend filter won't satisfy my need.

Because the playground filter won't auto-split my value when using equals operator,  if I type `1,2,3` , it will treat it like `= "1,2,3"` while generating it to SQL, so I tried another way which is calling the /load API with customized query by myself,
```
"filters":[{"member":"users.age","operator":"equals","values":["23","24","25"]}]
```
It works now, so the main problem is the filter of playground treats all the values as a single type.

Just call the API and pass the query you want directly, and you will get the expected result.
